### PR TITLE
ZK-3973: potentially locked resource files

### PR DESCRIPTION
**Pull request by scribetw**

### CODE REVIEW

File: zcommon/src/org/zkoss/util/resource/AbstractLoader.java [hunk 1]
```diff
@@ -18,34 +18,43 @@
...
+				conn = ((URL) src).openConnection();
+				final long v = conn.getLastModified();
+				return v != -1 ? v : 0; //not to reload if unknown (5.0.6 for better performance)
 			} catch (Throwable ex) {
 				return -1; //reload (might be removed)
+			} finally {
+				if (conn != null) {
+					try {
+						conn.getInputStream().close();
+					} catch (Exception ignored) { }
```
Comment (DevChu) [inline zcommon/src/org/zkoss/util/resource/AbstractLoader.java]: here should log something

File: zweb/src/org/zkoss/web/util/resource/ExtendletLoader.java [hunk 2]
```diff
@@ -75,13 +76,21 @@ public long getLastModified(String src) {
...
+				conn = url.openConnection();
+				final long v = conn.getLastModified();
 				return v != -1 ? v : 0; //not to reload (5.0.6 for better performance)
 			}
 		} catch (Throwable ex) {
+		} finally {
+			if (conn != null) {
+				try {
+					conn.getInputStream().close();
+				} catch (Exception ignored) { }
```
Comment (DevChu) [inline zweb/src/org/zkoss/web/util/resource/ExtendletLoader.java]: here should log something

File: zweb/src/org/zkoss/web/util/resource/ResourceLoader.java [hunk 3]
```diff
@@ -62,11 +63,19 @@ public boolean shallCheck(ResourceInfo src, long expiredMillis) {
...
+				conn = src.url.openConnection();
+				final long v = conn.getLastModified();
 				return v != -1 ? v : 0; //not to reload (5.0.6 for better performance)
 			} catch (Throwable ex) {
 				return -1; //reload (might be removed)
+			} finally {
+				if (conn != null) {
+					try {
+						conn.getInputStream().close();
+					} catch (Exception ignored) { }
```
Comment (DevChu) [inline zweb/src/org/zkoss/web/util/resource/ResourceLoader.java]: here should log something

**Comment by klyvechen** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Comment by klyvechen** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**